src/mesh.cpp

```diff
@@ -1597,6 +1621,108 @@ extern "C" int openmc_rectilinear_mesh_set_grid(int32_t index,
   return err;
 }
 
+//! Get the cylindrical mesh grid
+extern "C" int openmc_cylindrical_mesh_get_grid(int32_t index, double** grid_x,
+  int* nx, double** grid_y, int* ny, double** grid_z, int* nz)
+{
+  if (int err = check_mesh_type<CylindricalMesh>(index))
+    return err;
+  CylindricalMesh* m =
+    dynamic_cast<CylindricalMesh*>(model::meshes[index].get());
+
+  if (m->lower_left_.dimension() == 0) {
+    set_errmsg("Mesh parameters have not been set.");
+    return OPENMC_E_ALLOCATE;
+  }
+
+  *grid_x = m->grid_[0].data();
+  *nx = m->grid_[0].size();
+  *grid_y = m->grid_[1].data();
+  *ny = m->grid_[1].size();
+  *grid_z = m->grid_[2].data();
+  *nz = m->grid_[2].size();
+
+  return 0;
+}
+
+//! Set the cylindrical mesh parameters
+extern "C" int openmc_cylindrical_mesh_set_grid(int32_t index,
+  const double* grid_x, const int nx, const double* grid_y, const int ny,
+  const double* grid_z, const int nz)
+{
+  if (int err = check_mesh_type<CylindricalMesh>(index))
+    return err;
+  CylindricalMesh* m =
+    dynamic_cast<CylindricalMesh*>(model::meshes[index].get());
+
+  m->n_dimension_ = 3;
+  m->grid_.resize(m->n_dimension_);
+
+  for (int i = 0; i < nx; i++) {
```

Comment:
Should call grid_[x].reserve(nx), etc. before calling push_back many times.

Reply:
I do that, but these are only some 100 values at initialization, so not that bad
I personally dislike the C-style interface with all the pointers much less. I would port that to pybind11. Don't know, if this is still an idea or if somebody already is working on that.

Reply:
@ojschumann Yes, the C-style interfaces are a bit messy. Using pybind11 should improve things substantially -- I did have a student who was working on this a while ago but it never quite got finished. I'll see if I can find the branch that they had going. Are you interested in working on that?

Reply:
@paulromano Yes indeed, I'm interested in that topic. I use pybind11 in a handful of my projects, so I have quite some experience. If you find that branch, I would check if I can work from that on. But I can't work on that full time, so this might take a while.

Reply:
@ojschumann, it may be 100 values, but you're calling malloc 8 times (assuming 100 mesh points) as opposed to once if you used reserve. Malloc is really expensive. If we pay attention to this everywhere we initialize stuff in OpenMC, it adds up.